### DISCUSSION TITLE
AddPassKey email account

### DISCUSSION THREAD

**Discussion by nsnull0** [alchemyplatform/aa-sdk#906] [Ideas]
```
 try {
                let arr = await signerAA?.addPasskey()
            } catch (error) {
                ..
            }
```
I tried with this method to add passkey for email account
It is working fine, successfully return credentialIds.

would be great if getAuthDetails method can return credentialIds too when email account have add passkey.
currently, it always return null regardless I already add passkey.

**Comment by moldy530** [comment]
I was just thinking about this the other day and came across this. Yea it's definitely missing right now. Will let you know once I add this (when I get around to it, should hopefully be pretty easy)

**Reply by moldy530** [reply]
only thing is that when logging in with email, there might be multiple credentialIds added to a user's account (since they may add multiple credentials). This field on the user object is really just meant to be used for caching the last used passkey.

What's your use case for having the credentialIds when logging in with email?

**Reply by nsnull0** [reply]
Thank you for following up.. yea, i am aware that user can do multiple addPasskey.

I guess not a critical use case, its about to displaying a flag on setting page whether the user's account already has a passkey added.
so would be great IF i could get last used passkey or an array of credentialIds of particular account.
